airflow/www/views.py

```diff
@@ -714,34 +714,38 @@ def rendered(self):
     @login_required
     @wwwutils.action_logging
     def log(self):
-        BASE_LOG_FOLDER = os.path.expanduser(
-            conf.get('core', 'BASE_LOG_FOLDER'))
+        base_log_folder = os.path.expanduser(conf.get('core', 'BASE_LOG_FOLDER'))
         dag_id = request.args.get('dag_id')
         task_id = request.args.get('task_id')
         execution_date = request.args.get('execution_date')
+        dttm = dateutil.parser.parse(execution_date)
+        iso = dttm.isoformat()
         dag = dagbag.get_dag(dag_id)
-        log_relative = "{dag_id}/{task_id}/{execution_date}".format(
-            **locals())
-        loc = os.path.join(BASE_LOG_FOLDER, log_relative)
-        loc = loc.format(**locals())
-        log = ""
+        log_relative_dir = "{dag_id}/{task_id}/{iso}".format(**locals())
+        local_log_dir = os.path.join(base_log_folder, log_relative_dir)
+        local_log_dir = local_log_dir.format(**locals())
+        logs = []
         TI = models.TaskInstance
-        dttm = dateutil.parser.parse(execution_date)
         form = DateTimeForm(data={'execution_date': dttm})
         session = Session()
         ti = session.query(TI).filter(
             TI.dag_id == dag_id, TI.task_id == task_id,
             TI.execution_date == dttm).first()
 
+        def filename(directory, try_number):
+            return "{}/{}.log".format(directory, try_number+1)
+
         if ti is None:
-            log = "*** Task instance did not exist in the DB\n"
+            logs = ["*** Task instance did not exist in the DB\n"]
         else:
             # load remote logs
             remote_log_base = conf.get('core', 'REMOTE_BASE_LOG_FOLDER')
```

Comment:
Let's make sure to test these cases manually since we don't have integration tests for them.